core/src/main/scala/kafka/raft/KafkaNetworkChannel.scala

```diff
@@ -175,10 +175,10 @@ class KafkaNetworkChannel(time: Time,
 
   private def buildInboundRaftResponse(response: ClientResponse): RaftResponse.Inbound = {
     val header = response.requestHeader()
-    val data = if (response.wasDisconnected) {
-      errorResponseData(header.apiKey, Errors.BROKER_NOT_AVAILABLE)
-    } else if (response.authenticationException != null) {
+    val data = if (response.authenticationException != null) {
```

Comment:
Is this a bug fix? The authentication and disconnection check must have strict ordering?

Reply:
Yes, that's right. Currently when we get an authentication exception, we also set the disconnect flag in `ClientResponse`. So we have to check for the authentication failure first.